protocols/gossipsub/src/behaviour.rs

```diff
@@ -747,6 +749,19 @@ where
         let mut publish_failed = true;
         for peer_id in recipient_peers.iter() {
             tracing::trace!(peer=%peer_id, "Sending message to peer");
+            // If enabled, Send first an IDONTWANT so that if we are slower than forwarders
+            // publishing the original message we don't receive it back.
+            if raw_message.raw_protobuf_len() > self.config.idontwant_message_size_threshold()
+                && self.config.idontwant_on_publish()
+            {
+                self.send_message(
+                    *peer_id,
+                    RpcOut::IDontWant(IDontWant {
+                        message_ids: vec![msg_id.clone()],
+                    }),
+                );
+            }
+
```

Comment:
I assume you didn't use `send_idontwant` to have to aggregate and iter `recipient_peers` twice, right? 
Does it matter that the set of `recipient_peers` here differ from the `recipient_peers` in `send_idontwant`? E.g., we now don't just sent it to `PeerKind::Gossipsubv1_2` peers, but to all peers in the list.

Reply:
> E.g., we now don't just sent it to `PeerKind::Gossipsubv1_2` peers, but to all peers in the list.

Ah sorry, saw just now that you are filtering it out in `send_message`.